packages/block-editor/src/components/global-styles/background-panel.js

```diff
@@ -722,6 +869,75 @@ export default function BackgroundPanel( {
 			_links: _settings[ globalStylesLinksDataKey ],
 		};
 	}, [] );
+
+	const colors = useColorsPerOrigin( settings );
+	const gradients = useGradientsPerOrigin( settings );
+	const areCustomSolidsEnabled = settings?.color?.custom;
+	const areCustomGradientsEnabled = settings?.color?.customGradient;
+	const hasSolidColors = colors.length > 0 || areCustomSolidsEnabled;
+	const hasGradientColors = gradients.length > 0 || areCustomGradientsEnabled;
+	const encodeColorValue = ( colorValue ) => {
+		const allColors = colors.flatMap(
+			( { colors: originColors } ) => originColors
+		);
+		const colorObject = allColors.find(
+			( { color } ) => color === colorValue
+		);
+		return colorObject
+			? 'var:preset|color|' + colorObject.slug
+			: colorValue;
+	};
+	const encodeGradientValue = ( gradientValue ) => {
+		const allGradients = gradients.flatMap(
+			( { gradients: originGradients } ) => originGradients
+		);
+		const gradientObject = allGradients.find(
+			( { gradient } ) => gradient === gradientValue
+		);
+		return gradientObject
+			? 'var:preset|gradient|' + gradientObject.slug
+			: gradientValue;
+	};
+
+	const decodeValue = ( rawValue ) =>
+		getValueFromVariable( { settings }, '', rawValue );
+	const shouldShowBackgroundColorControls = useHasBackgroundPanel( settings );
```

Comment:
Hmm, I think there's an unfortunate naming clash. Wouldn't we be more interested in the function of the same same from the color panel?

https://github.com/WordPress/gutenberg/blob/9cd986ca8e7331cfb8302a85d1070a5bacf2992c/packages/block-editor/src/components/global-styles/color-panel.js#L100-L100

I think one or both of the functions names could be changes, e.g. `useHasBackgroundImagePanel` and `useHasBackgroundColorPanel`

I'll get a quick PR up for that 👍🏻 

In the mean time we could import the right function and rename it, e.g., `import { useHasBackgroundPanel as useHasBackgroundColorPanel } from './color-panel';`

Reply:
Renaming PR: https://github.com/WordPress/gutenberg/pull/64993